File: IrrgartenP3/IrrgartenJava/src/irrgarten/Shield.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author moham
 */
// Shield.java
public class Shield {
    private float protection;
    private int uses;

    public Shield(float protection, int uses) {
        this.protection = protection;
        this.uses = uses;
    }

    public float protect() {
        if (uses > 0) {
            uses--;
            return protection;
        } else {
            return 0;
        }
    }

    @Override
    public String toString() {
        return "S[" + protection + ", " + uses + "]";
    }

    public boolean discard() {
        return Dice.discardElement(uses); // Este método se implementará en la clase Dice.
    }
}

File: IrrgartenP3/irrgartenRuby/irrgarten/shield.rb
module Irrgarten
  class Shield
  attr_reader :protection, :uses

  def initialize(protection, uses)
    @protection = protection
    @uses = uses
  end

  def protect
    if @uses > 0
      @uses -= 1
      @protection
    else
      0
    end
  end

  def to_s
    "S[#{@protection}, #{@uses}]"
  end

  def discard
    Dice.discard_element(@uses)
  end
end
end